src/compiler/checker.ts

```diff
@@ -48500,6 +48576,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return factory.createToken(SyntaxKind.AnyKeyword) as KeywordTypeNode;
         }
         const signature = getSignatureFromDeclaration(signatureDeclaration);
+        const typePredicate = getTypePredicateOfSignature(signature);
+        if (typePredicate) {
+            const parameterName = typePredicate.kind === TypePredicateKind.Identifier || typePredicate.kind === TypePredicateKind.AssertsIdentifier ?
+                setEmitFlags(factory.createIdentifier(typePredicate.parameterName), EmitFlags.NoAsciiEscaping) :
+                factory.createThisTypeNode();
+            const typeNode = typePredicate.type && nodeBuilder.typeToTypeNode(typePredicate.type);
+            return factory.createTypePredicateNode(/*assertsModifier*/ undefined, parameterName, typeNode);

```

Comment:
Even though we only infer `Identifier` non-assert type predicates right now, it'd be nice if this was robust enough to print the other predicate kinds without morphing them into identifier predicates.

Reply:
Sure, done.